lib/spack/spack/cmd/develop.py

```diff
@@ -84,86 +79,150 @@ def _retrieve_develop_source(spec: spack.spec.Spec, abspath: str) -> None:
+        else:
+            # look up the maximum version so infintiy versions are preferred for develop
+            version = max(spec.package_class.versions.keys())
+            tty.msg(f"Defaulting to highest version: {spec.name}@{version}")
+    spec.versions = spack.version.VersionList([version])
 
-            # Both old syntax `spack develop pkg@x` and new syntax `spack develop pkg@=x`
-            # are currently supported.
-            spec = spack.spec.parse_with_version_concrete(entry["spec"])
-            _retrieve_develop_source(spec, abspath)
 
-        if not env.dev_specs:
-            tty.warn("No develop specs to download")
+def setup_src_code(spec: spack.spec.Spec, src_path: str, clone: bool = True, force: bool = False):
+    """
+    Handle checking, cloning or overwriting source code
+    """
+    assert spec.versions
+    if clone is None:
+        clone = not os.path.exists(src_path)
 
-        return
+    if clone:
+        _clone(spec, src_path, force)
 
-    specs = spack.cmd.parse_specs(args.spec)
-    if len(specs) > 1:
-        raise SpackError("spack develop requires at most one named spec")
+    if not clone and not os.path.exists(src_path):
+        raise SpackError(f"Provided path {src_path} does not exist")
 
-    spec = specs[0]
 
-    version = spec.versions.concrete_range_as_version
-    if not version:
-        # look up the maximum version so infintiy versions are preferred for develop
-        version = max(spec.package_class.versions.keys())
-        tty.msg(f"Defaulting to highest version: {spec.name}@{version}")
-    spec.versions = spack.version.VersionList([version])
+def _update_config(spec, path):
+    find_fn = lambda section: spec.name in section
 
-    # If user does not specify --path, we choose to create a directory in the
-    # active environment's directory, named after the spec
-    path = args.path or spec.name
-    if not os.path.isabs(path):
-        abspath = spack.util.path.canonicalize_path(path, default_wd=env.path)
-    else:
-        abspath = path
+    entry = {"spec": str(spec)}
+    if path != spec.name:
+        entry["path"] = path
 
-    # clone default: only if the path doesn't exist
-    clone = args.clone
-    if clone is None:
-        clone = not os.path.exists(abspath)
+    def change_fn(section):
+        section[spec.name] = entry
 
-    if not clone and not os.path.exists(abspath):
-        raise SpackError("Provided path %s does not exist" % abspath)
+    spack.config.change_or_add("develop", find_fn, change_fn)
 
-    if clone:
-        if os.path.exists(abspath):
-            if args.force:
-                shutil.rmtree(abspath)
-            else:
-                msg = "Path %s already exists and cannot be cloned to." % abspath
-                msg += " Use `spack develop -f` to overwrite."
-                raise SpackError(msg)
 
-        _retrieve_develop_source(spec, abspath)
+def update_env_file(
```

Comment:
I think this would be easier to read as `update_env` since it's interacting with the `Environment` through the normal `spack.config` interface, not the `EnvironmentManifestFile` interface